.circleci/config.yml

```diff
@@ -1,54 +1,56 @@
 version: 2
-# Required secret environment variables
-# AWS_ACCESS_KEY_ID
-# AWS_SECRET_ACCESS_KEY
 
 jobs:
-  test:
+  all:
     docker:
       - image: circleci/node:10-stretch
     steps:
       - checkout
       - run:
           name: Install dependencies
-          command: npm i
+          command: |
+            npm i
+            npm run bootstrap:ci
       - run:
           name: Lint
           command: npm run lint
+      - run:
+          name: Build
+          command: npm run build
       - run:
           name: Test
           command: npm run test
+      - run:
+          name: Configure deployment
+          command: |
+            static_s3_base="s3://cf-apps-static/apps-test-$CIRCLE_SHA1"
+            test "$CIRCLE_BRANCH" == 'master' && static_s3_base='s3://cf-apps-static/apps'
+            echo "export STATIC_S3_BASE=\"$static_s3_base\"" >> $BASH_ENV
 
-  deploy:
-    docker:
-      - image: circleci/node:10-stretch
-        environment:
-          STATIC_APPS_BUCKET: s3://cf-apps-static
-    steps:
-      - checkout
+            static_jira_s3_base="s3://cf-apps-static/apps-test-$CIRCLE_SHA1/jira"
+            test "$CIRCLE_BRANCH" == 'master' && static_jira_s3_base='s3://cf-apps-jira'
+            echo "export STATIC_JIRA_S3_BASE=\"$static_jira_s3_base\"" >> $BASH_ENV
+
+            stage='test'
```

Comment:
I disagree with the fact that we need to have lambda deployment run on every branch. Especially with this setup, in which weird  side effects will most certainly occur when deployment is run on a single stage from multiple branches.

Reply:
I think benefits outweigh problems. Right now the first time we attempt to deploy from CI to Lambda is after we merge to master. This blocks further deployments until the problem is solved (likely: CI role cannot do something) or the change is reverted. By attempting to deploy in branches we mitigate this risk.

Reply:
As long as we just block master, I think this is fine. In case we need to ship something else fast, we can still revert the changes. If we're worried that the prd deployment might actually break something important, I'd suggest we add a dev deployment stage in the master build.

Reply:
Do you mean merge to master = staging, merge to `prod` branch = deployment to production?

Reply:
Other issue with the current state of affairs: frontend gets deployed, backend deployment fails. Frontend relies on new backend version. App is broken.

Reply:
I mean that the master build would be something like
1) deploy to `dev`
2) smoke test dev (:smiling_imp:)
3) deploy to `prd`

Frontend and backend will always have to remain compatible between individual changes, since we cannot guarantee an instant client side code update due to caching anyway. We also cannot gurantee the order of deployment.

I'd also like to avoid the `prod` branch approach. I think we both agreed that it is a very problematic solution.